app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -454,6 +455,10 @@ public void ensureEditIsFunctioning() {
         swipeLeftOnHabit(testHabit);
         _solo.waitForView(R.id.edit_button);
         _solo.clickOnButton(EDIT_BUTTON);
+
+        // wait for edit habit dialog to spawn after edit is clicked
+        _solo.waitForText(EDIT_HABIT_DIALOG);
+
```

Comment:
Added since Robotium would not wait for the Edit Dialog to pop up

Reply:
Can we set a **timeout** to this, instead of it just hanging up on this line in the case where it **doesn't** show up? Usually, the timeout is the second parameter in the method, in milliseconds. If possible, let's set it to something under 5 seconds (<5000)